api/src/main/clojure/core2/datalog.clj

```diff
@@ -11,7 +14,9 @@
        (reify Function
          (apply [_ res]
            (with-open [^IResultSet res res]
-             (vec (iterator-seq res))))))))
+             (->> (iterator-seq res)
```

Comment:
calling `keyword` is relatively expensive here, per-row. `IResultSet` has `.columnTypes`, from the keys of which you can generate a mapping and use that as your function in `update-keys`, and you'll only be creating the keyword objects once.